dialplan/asterisk/extensions_lib_pre_dial.conf

```diff
@@ -0,0 +1,19 @@
+; Copyright 2021 The Wazo Authors  (see the AUTHORS file)
+; SPDX-License-Identifier: GPL-2.0+
+
+[wazo-add-pre-dial-hook]
+exten = s,1,Set(HANDLER=${ARG1})
+same = n,Set(PUSH(_WAZO_PRE_DIAL_HANDLERS,|)=${HANDLER}^s^1)
+same = n,Return()
+
+[wazo-pre-dial-hooks]
+exten = s,1,While($["${SET(pre-dial-handler=${POP(WAZO_PRE_DIAL_HANDLERS,|)})}" != ""]) ; or SHIFT, if we want FIFO execution
+same = n,GoSub(${REPLACE(pre-dial-handler,^,,)})
+same = n,EndWhile()
+same = n,Return()
+
+[wazo-schedule-pre-dial-hooks]
+exten = s,1,AGI(agi://${XIVO_AGID_IP}/convert_b_option)
+same = n,GotoIf(${WAZO_PRE_DIAL_HANDLERS}?:done)
```

Comment:
How useful is this `GotoIf`? If we remove it, the `While` loop will simply do nothing, right?

Reply:
I wanted to avoid paying the price of jumping back to the dialplan to execute the `b` when there's nothing to execute. That said, I have no idea if that price is high or not.

Reply:
Oh I see, it completely removes the pre-dial logic from the future channel :) Let's keep the `GotoIf` then.